### PR TITLE
Added partition-level override to cell data

### PR DESCRIPTION
**Pull request by bretttegartms**
We can use this to customize attributes like instance size or type per partition.

### CODE REVIEW

File: lib/python/treadmill_aws/autoscale.py [hunk 1]
```diff
@@ -123,22 +134,29 @@ def create_n_servers(count, partition=None, pool=None):
     admin_srv = context.GLOBAL.admin.server()
     admin_cell = context.GLOBAL.admin.cell()
     cell = context.GLOBAL.cell
-    data = admin_cell.get(cell)['data']
+    cell_data = admin_cell.get(cell)['data']
+
+    admin_part = admin.Partition(context.GLOBAL.ldap.conn)
```
[the remark this answers is not in the document]
Comment (bretttegartms) [reply lib/python/treadmill_aws/autoscale.py]: Corrected.

File: lib/python/treadmill_aws/autoscale.py [hunk 2]
```diff
@@ -109,6 +111,15 @@ def _split_list(lst, num_parts):
     return [part for part in parts if part]
 
 
+def _get_data(cell_data, partition_data, name):
+    """Return partion data if set, Cell data if not. If both missing, False"""
+    if partition_data.get(name, None):
```
[the remark this answers is not in the document]
Comment (bretttegartms) [reply lib/python/treadmill_aws/autoscale.py]: Right, makes sense to fail early and noisily. Updated to reflect your suggestion.

[the remark this answers is not in the document]
Comment (mswiecicki-ms) [reply lib/python/treadmill_aws/autoscale.py]: _get_data is no longer used, right?

[the remark this answers is not in the document]
Comment (bretttegartms) [reply lib/python/treadmill_aws/autoscale.py]: Oops, you're right. Removed unneeded function.

**Pull request merged**